src/resolve-config.js

```diff
@@ -3,31 +3,39 @@
 const cosmiconfig = require("cosmiconfig");
 const minimatch = require("minimatch");
 
-const withCache = cosmiconfig("prettier");
-const noCache = cosmiconfig("prettier", { cache: false });
+const asyncWithCache = cosmiconfig("prettier");
```

Comment:
Although I am not a contributor to `prettier`, I have a minor suggestion. Wouldn't it be better to avoid creation of multiple `cosmiconfig` objects? I also thought of a way to do:

```js
const makeKey = (sync, useCache) => `sync_${sync}|useCache_${useCache}`;

const configLoaders = new Map();

function configLoaderFor(opts) {
  const sync = opts && opts.sync === true;
  const useCache = !(opts && opts.useCache === false);

  const key = makeKey(sync, useCache);

  if (configLoaders.has(key)) return configLoaders.get(key);

  const configLoader = cosmiconfig('prettier', { cache: useCache, sync });
  configLoaders.set(key, configLoader);
  return configLoader;
}
```

Also I would suggest to use the [`transform` option](https://github.com/davidtheclark/cosmiconfig#transform) than manually invoking `mergeOverrides`.

Reply:
There was a reason I didn't use `transform`. I think it was so we can safely call `resolveConfig()` on the same file with different sets of options without it being cached.